lib/platform/linux/platform.c

```diff
@@ -207,16 +244,17 @@ static void * poll_for_indication(void * unused)
                 free_buffer_room = m_ind_queue_read - m_ind_queue_write;
             }
         }
-
-        max_num_indication = free_buffer_room > MAX_NUMBER_INDICATION ?
-                                 MAX_NUMBER_INDICATION :
-                                 free_buffer_room;
+                  
+        max_num_indication = (m_polling_thread_state_request == POLLING_THREAD_STOP_REQUESTED)
```

Comment:
Maybe as it is now a mix of 2 test it would be easier to read with a standard if that allow to add more comments :-) 
```

if (m_polling_thread_state_request == POLLING_THREAD_STOP_REQUESTED)
{
    // In case we are about to stop, let's poll only one by one to have more chance to finish
    // uncomplete fragmented packet and not start to receive a new one
    max_num_indication = 1;
}
else
{
 ....
}
```